R/assert.R

```diff
@@ -178,20 +163,129 @@ vec_is <- function(x, ptype = NULL, size = NULL) {
   TRUE
 }
 
-#' Is object a vector?
-#' @noRd
+#' Vector checks
 #'
 #' @description
 #'
-#' Returns `TRUE` if:
+#' - `vec_is_vector()` tests if `x` is considered a vector in the vctrs sense.
```

Comment:
I felt it was ok as a `vec_` function because we also have `vec_is_list()` which also _seems_ to imply that you can't pass scalars, but you can

``` r
vec_is_list(environment())
#> [1] FALSE
```

Reply:
hmm I'm surprised that we can.

Reply:
Reading it again I feel very positively about this naming scheme. I like that a prefixed predicate implies a notion of type narrowing: from object to vector, or from vector to a vector of a certain size.

Reply:
Me too!

Reply:
And I think we've mostly gotten this "right" everywhere besides `vec_is_list()` and `vec_check_list()`.

Like `vec_is_empty()` _does_ require vector input, so that name is right.

And it makes it more obvious that `vec_assert()` and `vec_is()` aren't great names